### PR TITLE
feat: add assert util

### PR DESCRIPTION
**Pull request by RomainLanz**
Hey there! 👋🏻 

This PR adds some `assert` utils to help code flow and type-safety in the codebase.

### CODE REVIEW

**Comment by thetutlage** [comment]
Should we export it from `@poppinss/utils/assert`?

**Comment by RomainLanz** [comment]
Oh yeah, we can also add an export from here. Let me update the PR.

File: index.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@
 import { fileURLToPath } from 'node:url'
 import { dirname as pathDirname } from 'node:path'
 
+export { assert, assertNotNull, assertUnreachable } from './src/assert.js'
```
Comment (thetutlage) [inline index.ts]: Remove it from here too?

Comment (RomainLanz) [reply index.ts]: Done!

**Pull request merged**